### PR TITLE
[9.0] document_page

### PR DESCRIPTION
**Pull request by naousse**

### CODE REVIEW

File: document_page/readme.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+Module's migration
```
Comment (pedrobaeza) [inline document_page/readme.md]: This should be in RST format with name README.rst and follow this template: https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

**Comment by naousse** [comment]
OK. I will do it

File: document_page/__openerp__.py [hunk 2]
```diff
@@ -21,7 +21,7 @@
 
 {
     'name': 'Document Page',
-    'version': '8.0.1.0.1',
+    'version': '9.0.1.0.0',
     'category': 'Knowledge Management',
     'description': """
```
Comment (dreispt) [inline document_page/__openerp__.py]: Remove the description key

File: document_page/__openerp__.py [hunk 3]
```diff
@@ -32,23 +32,22 @@
     'website': 'http://www.openerp.com/',
```
Comment (dreispt) [inline document_page/__openerp__.py]: Add OCA in the authors

File: document_page/__openerp__.py [hunk 4]
```diff
@@ -32,23 +32,22 @@
...
-        'document_page_demo.xml'
+        'demo/document_page.xml'
     ],
     'test': [
         'test/document_page_test00.yml'
     ],
-    'installable': False,
+    'installable': True,
     'auto_install': False,
     'images': [],
```
Comment (dreispt) [inline document_page/__openerp__.py]: AFAIK images is deprecated

**Comment by naousse** [comment]
@dreispt @pedrobaeza , please i have fixed some errors that were in my pull request. Please can you check now if the module are OK.

File: document_page/README.rst [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+==============
+Document Page
+==============
+
+This module allows you to write web pages for internal documentation.
+
+Installation
+============
+
+This module depends on module knowledge. So make sure to have available it in your addons list
```
Comment (max3903) [inline document_page/README.rst]: ... make sure to have it in your addons list.

File: document_page/README.rst [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+Usage
+=====
+
+To use this module, you need to:
+* Install the module's dependencies
+* Install this module module
+* Got to knowledge menu
+* use submenu Categories to Create the document's category you need
+* use submenu Pages to  Create pages for that category
```
Comment (max3903) [inline document_page/README.rst]: To use this module, you need to:
- Go to Knowledge menu
- Click on Categories to create the document's category you need with the template
- Click on Pages to create pages and select the previous category to use the template

File: document_page/README.rst [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+* use tu submenu Pages history for pages revision history
+
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/{repo_id}/{branch}
+
+.. repo_id is available in https://github.com/OCA/maintainer-tools/blob/master/tools/repos_with_ids.txt
+.. branch is "8.0" for example
+
```
Comment (max3903) [inline document_page/README.rst:34]: The target link should be https://runbot.odoo-community.org/runbot/118/9.0

You can also remove the comments.

File: document_page/README.rst [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/
+{project_repo}/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback `here <https://github.com/OCA/
+{project_repo}/issues/new?body=module:%20
+{module_name}%0Aversion:%20
+{version}%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
+
```
Comment (max3903) [inline document_page/README.rst]: Replace {project_repo} , {module_name} and {version}, by knowledge, document_page and 9.0 respectively.

File: knowledge/README.rst [hunk 9]
```diff
@@ -0,0 +1,104 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+================================
+Knowledge Management System base
+================================
+
+This module allows you to write web pages for internal documentation.
+
```
Comment (max3903) [inline knowledge/README.rst:10]: This module provides the base for the Knowledge applications (menu, groups, configuration panel).

File: knowledge/README.rst [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+Installation
+============
+
+Makes the Knowledge Application Configuration available from where you can install
+document and Wiki based Hidden.
```
Comment (max3903) [inline knowledge/README.rst]: No specific installation step required.

File: knowledge/README.rst [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+Configuration
+=============
+
+No configuration required
+
+Usage
+=====
+
+To use this module, you need to:
+
```
Comment (max3903) [inline knowledge/README.rst:25]: - go to Knowledge > Configuration > Settings to install other applications.

File: knowledge/README.rst [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/{repo_id}/{branch}
```
Comment (max3903) [inline knowledge/README.rst]: https://runbot.odoo-community.org/runbot/118/9.0

File: knowledge/README.rst [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/
+{project_repo}/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback `here <https://github.com/OCA/
+{project_repo}/issues/new?body=module:%20
+{module_name}%0Aversion:%20
+{version}%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
+
```
Comment (max3903) [inline knowledge/README.rst:48]: Same as before but with:
knowledge
knowledge
9.0

File: document_page/views/document_page.xml [hunk 14]
```diff
@@ -1,8 +1,8 @@
 <?xml version="1.0"?>
 <openerp>
     <data>
-       <menuitem name="Knowledge"
-                 id="knowledge.menu_document" />
+      <menuitem name="Knowledge"
+                id="knowledge.menu_document" />
```
Comment (max3903) [inline document_page/views/document_page.xml]: No need for this. This menu item is already created here:
https://github.com/OCA/knowledge/pull/76/files#diff-8186cbed73afd58aa336582b1d4c5ae1R6

File: knowledge/README.rst [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+To use this module, you need to:
+
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/{repo_id}/{branch}
+
+.. repo_id is available in https://github.com/OCA/maintainer-tools/blob/master/tools/repos_with_ids.txt
+.. branch is "8.0" for example
+
```
Comment (max3903) [inline knowledge/README.rst:32]: Remove those 2 comments lines.

File: document_page/README.rst [hunk 16]
```diff
@@ -0,0 +1,94 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+==============
+Document Page
+==============
```
Comment (max3903) [inline document_page/README.rst]: @naousse Please remove the extra =

**Comment by max3903** [comment]
@naousse Please keep the knowledge module for #75

**Comment by naousse** [comment]
I removed the extra = in readme.rst and the knowledge module

File: document_page/models/document_page_history.py [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+
+
+class DocumentPageHistory(models.Model):
+    _name = "document.page.history"
+    _description = "Document Page History"
+    _order = 'id DESC'
+    _rec_name = "create_date"
+
+    page_id = fields.Many2one('document.page', 'Page')
+    summary = fields.Char('Summary', size=256, select=True)
```
Comment (dreispt) [inline document_page/models/document_page_history.py]: Remove size

**Comment by max3903** [comment]
@naousse Needs rebasing and fixing ;)

File: document_page/README.rst [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+Configuration
+=============
+
+No configuration required
+
+Usage
+=====
+
+To use this module, you need to:
+* Go to Knowledge menu
```
Comment (max3903) [inline document_page/README.rst:26]: @naousse You need an extra line before the list, otherwise rendering fails.

File: document_page/README.rst [hunk 19]
```diff
@@ -0,0 +1,90 @@
...
+promote its widespread use.
+
+To contribute to this module, please visit http://odoo-community.org.
+
+Changelog
+---------
+
+v 9.0.1.0.0
+
+Creation of folders models, views, data and demo
```
Comment (max3903) [inline document_page/README.rst]: @naousse Needs some cleanup.

**Comment by max3903** [comment]
Tested on runbot
:+1:

File: document_page/data/document_page.xml [hunk 20]
```diff
@@ -8,8 +8,8 @@
             <field name="index">1</field>
             <field name="summary">Initial Page</field>
             <field  name="content">==The OpenERP wiki==
-            	
-[[File:http://www.openerp.com/sites/all/themes/openerp/logo.png OpenERP]]            	
+
+[[File:http://www.openerp.com/sites/all/themes/openerp/logo.png OpenERP]]
```
Comment (dreispt) [inline document_page/data/document_page.xml]: Can you check this URL?

File: document_page/models/document_page.py [hunk 21]
```diff
@@ -144,51 +151,9 @@ def write(self, vals):
     @api.model
     @api.returns('self', lambda value: value.id)
     def create(self, vals):
-        page_id = super(document_page, self).create(vals)
+        """Create the first history of a document."""
```
Comment (dreispt) [inline document_page/models/document_page.py:153]: A few lines below, shouldn't `self.create_history(...` rather be `result.create_history(...`

File: document_page/models/document_page_history.py [hunk 22]
```diff
@@ -0,0 +1,72 @@
+"""DocumentPageHistory model."""
```
Comment (dreispt) [inline document_page/models/document_page_history.py]: It's not advised to the docstrings on python files. Add to Model classes instead.

File: document_page/models/document_page_history.py [hunk 23]
```diff
@@ -0,0 +1,72 @@
...
+    def __getattr__(self, attr):
+        """Return a dummy callabale."""
+        if attr in ['_sql', 'init', '_ids']:
+            raise AttributeError
+
+        _logger.warning(
+            "Trying to access attribute %s on document_page_history",
+            attr
+        )
+        return (lambda *a, **b: None)
```
Comment (dreispt) [inline document_page/models/document_page_history.py:71]: I wonder why this is needed. Any idea @max3903 ?

File: document_page/wizard/__init__.py [hunk 24]
```diff
@@ -1,3 +1,4 @@
+"""wizard's models registration."""
```
Comment (dreispt) [inline document_page/wizard/__init__.py]: Please remove

File: document_page/wizard/document_page_create_menu.py [hunk 25]
```diff
@@ -1,3 +1,4 @@
+"""DocumentPageCreateMenu model."""
```
Comment (dreispt) [inline document_page/wizard/document_page_create_menu.py]: idem

**Comment by dreispt** [comment]
Other than some minor notes, it has my :+1:

File: document_page/models/document_page_history.py (hunk 23 above)
Comment (max3903) [inline document_page/models/document_page_history.py:71]: @dreispt I don't know. Odoo SA made this module originally and they deprecated it in v8.

File: document_page/README.rst [hunk 26]
```diff
@@ -0,0 +1,69 @@
...
+   :target: https://runbot.odoo-community.org/runbot/118/9.0
+
+Known issues / Roadmap
+======================
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/
+{project_repo}/issues>`_.
```
Comment (max3903) [inline document_page/README.rst]: Replace `{project_repo}` with `knowledge`

File: document_page/data/document_page.xml [hunk 27]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<openerp>
+<odoo>
     <data noupdate="1">
         <record id="wiki_wiki_main" model="document.page">
             <field name="name">The OpenERP wiki</field>
```
Comment (max3903) [inline document_page/data/document_page.xml]: `The Odoo wiki`

File: document_page/data/document_page.xml (hunk 20 above)
Comment (max3903) [inline document_page/data/document_page.xml]: @naousse Please replace with https://www.odoo.com/openerp_website/static/src/img/logo_transparent_198px.png

File: document_page/data/document_page.xml [hunk 28]
```diff
@@ -8,8 +8,8 @@
             <field name="index">1</field>
             <field name="summary">Initial Page</field>
             <field  name="content">==The OpenERP wiki==
```
Comment (max3903) [inline document_page/data/document_page.xml]: `== The Odoo wiki ==`

File: document_page/data/document_page.xml [hunk 29]
```diff
@@ -8,8 +8,8 @@
...
-            	
-[[File:http://www.openerp.com/sites/all/themes/openerp/logo.png OpenERP]]            	
+
+[[File:http://www.openerp.com/sites/all/themes/openerp/logo.png OpenERP]]
 
 The OpenERP wiki allows you to manage your enterprise's contents using wiki
```
Comment (max3903) [inline document_page/data/document_page.xml]: `The Odoo wiki...`

**Comment by max3903** [comment]
@naousse Please take @dreispt suggestions to fix the module.

A screenshot with its path in the `images` keyword of the `__openerp__.py` would be nice, like here:
https://github.com/odoo/odoo/blob/9.0/addons/account/__openerp__.py#L17

File: document_page/models/document_page.py (hunk 21 above)
Comment (naousse) [inline document_page/models/document_page.py:153]: Hi @dreispt. 
Please i don't understand this remark. There is no result variable in this method. Please explain me what you talking about here

Comment (dreispt) [inline document_page/models/document_page.py:153]: Forget it.

File: document_page/models/document_page_history.py (hunk 23 above)
Comment (naousse) [inline document_page/models/document_page_history.py:71]: Even me i don't understand if this is needed. Can i removed it?

Comment (dreispt) [inline document_page/models/document_page_history.py:71]: Please do that.

**Comment by max3903** [comment]
:+1:

**Pull request merged**